astropy/cosmology/core.py

```diff
@@ -118,15 +119,11 @@ def __init_subclass__(cls):
         cls._init_signature = sig  # store (immutable) initialization signature
 
     def __new__(cls, *args, **kwargs):
-        # bundle initialization argument
+        self = super().__new__(cls)
+        # bundle and store initialization arguments on the instance
         ba = cls._init_signature.bind_partial(*args, **kwargs)
         ba.apply_defaults()  # and fill in the defaults
-        # get dictionary of arguments
-        parameters = ba.arguments
-
-        self = super().__new__(cls)
-        self._init_arguments = parameters  # store arguments from initialization
-
+        self._init_arguments = MappingProxyType(ba.arguments)
```

Comment:
I'm a bit confused about the rationale for protecting what seems to be a private attribute. Is it exposed elsewhere?

Reply:
Partly it's because I accidentally modified a value it held and confused myself in another PR. And partly because I've been trying to figure out how to position Cosmology to be more compatible with `astropy.modeling` and exposing ``_init_arguments`` will probably be part of the solution. Mostly it's because it ``_init_signature`` and ``_init_arguments`` are the abstract parameterizing of an immutable object and are never checked for correctness. ``_init_signature`` is immutable by the design of ``inspect.Signature``, and it's probably good to do the same for ``_init_arguments``.